src/main/java/net/dv8tion/jda/api/entities/Guild.java

```diff
@@ -2508,13 +2513,17 @@ default Task<List<Member>> findMembersWithRoles(@Nonnull Collection<Role> roles)
         for (Role role : roles)
             Checks.check(this.equals(role.getGuild()), "All roles must be from the same guild!");
 
-        if (isLoaded() || roles.isEmpty() || roles.contains(getPublicRole())) // Member#getRoles never contains the public role
+        if (isLoaded())
         {
             CompletableFuture<List<Member>> future = CompletableFuture.completedFuture(getMembersWithRoles(roles));
             return new GatewayTask<>(future, () -> {});
         }
 
-        return findMembers(member -> member.getRoles().containsAll(roles));
+        List<Role> rolesWithoutPublicRole = roles.stream().filter(role -> !role.isPublicRole()).collect(Collectors.toList());
+        if (rolesWithoutPublicRole.isEmpty())
+            return loadMembers();
```

Comment:
Originally this was returning a completed GatewayTask. Why is this doing loadMembers instead?

Reply:
When using `MemberCachePolicy.NONE` `findMembersWithRoles(publicRole)` would previously just return an empty Collection instead of a Collection that contains all Members

see https://github.com/DV8FromTheWorld/JDA/pull/2297#issuecomment-1288691702 and https://github.com/DV8FromTheWorld/JDA/pull/2297#issuecomment-1288754380